src/common.js

```diff
@@ -30,16 +31,36 @@ export function original(value) {
     // otherwise return undefined
 }
 
-export const assign =
-    Object.assign ||
-    function assign(target, value) {
-        for (let key in value) {
-            if (has(value, key)) {
-                target[key] = value[key]
+function assignMap(target, overrides) {
+    overrides.forEach(function(override) {
+        for (let key in override) {
+            if (has(override, key)) {
+                target.set(key, override[key])
             }
         }
-        return target
+    })
+    return target
+}
+function assignObjectLegacy(target, overrides) {
+    overrides.forEach(function(override) {
+        for (let key in override) {
+            if (has(override, key)) {
+                target[key] = override[key]
+            }
+        }
+    })
+    return target
+}
+export function assign(target) {
+    const overrides = Array.prototype.slice.call(arguments, 1)
```

Comment:
You can do `assign(target, ...overrides)` here

Reply:
Wouldn't it fail for ES5 environments?

Reply:
Immer is compiled to ES5 by [bili](https://github.com/egoist/bili), so no. ;)